src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/SnowflakeDataSourceItemFixture.cs

```diff
@@ -1,35 +1,85 @@
-﻿using Reveal.Sdk.Dom.Data;
+﻿using Reveal.Sdk.Dom.Core.Extensions;
+using Reveal.Sdk.Dom.Data;
 using Xunit;
 
 namespace Reveal.Sdk.Dom.Tests.Data.DataSourceItems
 {
     public class SnowflakeDataSourceItemFixture
     {
         [Fact]
-        public void ProcessDataOnServer_Should_GetAndSetProperties()
+        public void Constructor_ShouldSetTitleAndDataSource_WhenConstructedWithSnowflakeDataSource()
         {
             // Arrange
+            var title = "Test Item";
             var dataSource = new SnowflakeDataSource();
-            var dataSourceItem = new SnowflakeDataSourceItem("Test", dataSource);
 
             // Act
-            dataSourceItem.ProcessDataOnServer = true;
-            var processDataOnServer = dataSourceItem.ProcessDataOnServer;
+            var item = new SnowflakeDataSourceItem(title, dataSource);
 
             // Assert
-            Assert.True(processDataOnServer);
+            Assert.Equal(title, item.Title);
+            Assert.Equal(dataSource, item.DataSource);
```

Comment:
Should use [Theory]
Need more Asserts

DataSourceId = dataSource.id
title = DataSouce.Title if DataSource.Title == null